### PR TITLE
Add Sodium

### LINKED ISSUE #258: Add Sodium detection
### Some reference to the SDK, such as the developer site or a wikipedia article.

Sodium: https://doc.libsodium.org/

### SteamDB links

https://steamdb.info/app/1241950/
https://steamdb.info/app/1255460/

### Possible way to detect it

Contain any of the file: libsodium.dll, libsodium.so, libsodium.dylib

### PR DESCRIPTION
**Pull request by diedyesterdaywashere**
### SteamDB app page links to a few games using this

https://steamdb.info/app/1241950/
https://steamdb.info/app/1255460/

### Brief explanation of the change

closes #258 
Added a rule for Sodium

### CODE REVIEW

File: rules.ini [hunk 1]
```diff
@@ -186,6 +186,7 @@ OpenXR = (?:^|/)(?:lib)?openxr_loader\.(?:dll|so)$
 Photon = (?:^|/)Photon(?:3Unity3D|3DotNet)\.dll$
 Qt = (?:^|/)Qt(?:Core4|[0-9]Core)\.dll$
 SDL = (?:^|/)sdl2?\.dll$
+Sodium = (?:^|/)libsodium\.(?:dll|so|dylib)$
```
Comment (Lifeismana) [inline rules.ini:189]: you can just check for .dll ; .so & .dylib don't catch anything that doesn't get caught by .dll

Comment (diedyesterdaywashere) [reply rules.ini:189]: i just used all files that were listed in issue, but i can change it

**Comment by Lifeismana** [comment]
you forgot to remove .so & .dylib from non matching tests

**Comment by Lifeismana** [comment]
Looks good to me now, thanks for the PR!

**Pull request merged**